pretix_zugferd/invoice.py

```diff
@@ -325,30 +327,47 @@ def _zugferd_generate_document(self, invoice):
         doc.trade.settlement.monetary_summation.due_amount = total
         return doc
 
-    def generate(self, invoice):
-        fname, ftype, content = super().generate(invoice)
+    def _on_first_page(self, canvas: Canvas, doc):
+        super()._on_first_page(canvas, doc)
+        if not self.event.settings.zugferd_hide_label:
+            canvas.saveState()
+            canvas.translate(10 * mm, 10 * mm)
+            canvas.rotate(90)
+            canvas.setFont("OpenSansBd", 8)
+            canvas.setFillColorRGB(80 / 255, 161 / 255, 103 / 255)
+            canvas.drawString(
+                0, 0, _("eInvoice included")
+            )
+            canvas.restoreState()
 
+    def generate(self, invoice):
+        self.__zugferd = True
         if (
-            not invoice.invoice_from_name
-            or not invoice.invoice_to_country
-            or not invoice.invoice_from_country
+                not invoice.invoice_from_name
+                or not invoice.invoice_to_country
+                or not invoice.invoice_from_country
         ):
-            return fname, ftype, content
-
-        if str(invoice.invoice_to_country) in settings.COUNTRIES_OVERRIDE:
+            self.__zugferd = False
+        elif str(invoice.invoice_to_country) in settings.COUNTRIES_OVERRIDE:
             # It's not possible to generate a valid ZUGFeRD invoice with the recipient in a non-standard country code (e.g. Kosovo)
-            return fname, ftype, content
+            self.__zugferd = False
 
         try:
             xml = self._zugferd_generate_document(invoice).serialize(
```

Comment:
Should this be somehow fenced inside an `if self.__zugferd:`? Why try to generate the xml when we won’t attach it to the PDF anyway?